NullPointerException

Hi,

I encounter the following error while generating mappers:

```
An annotation processor threw an uncaught exception.
Consult the following stack trace for details.
java.lang.NullPointerException
	at fr.xebia.extras.selma.codegen.CustomMapperWrapper.collectCustomMethods(CustomMapperWrapper.java:200)
	at fr.xebia.extras.selma.codegen.CustomMapperWrapper.collectCustomMappers(CustomMapperWrapper.java:173)
	at fr.xebia.extras.selma.codegen.CustomMapperWrapper.<init>(CustomMapperWrapper.java:64)
	at fr.xebia.extras.selma.codegen.MapperWrapper.<init>(MapperWrapper.java:93)
	at fr.xebia.extras.selma.codegen.MapperClassGenerator.<init>(MapperClassGenerator.java:60)
	at fr.xebia.extras.selma.codegen.MapperProcessor.generateMappingClassses(MapperProcessor.java:89)
	at fr.xebia.extras.selma.codegen.MapperProcessor.process(MapperProcessor.java:74)
	at com.sun.tools.javac.processing.JavacProcessingEnvironment.callProcessor(JavacProcessingEnvironment.java:794)
	at com.sun.tools.javac.processing.JavacProcessingEnvironment.discoverAndRunProcs(JavacProcessingEnvironment.java:705)
	at com.sun.tools.javac.processing.JavacProcessingEnvironment.access$1800(JavacProcessingEnvironment.java:91)
	at com.sun.tools.javac.processing.JavacProcessingEnvironment$Round.run(JavacProcessingEnvironment.java:1035)
	at com.sun.tools.javac.processing.JavacProcessingEnvironment.doProcessing(JavacProcessingEnvironment.java:1176)
	at com.sun.tools.javac.main.JavaCompiler.processAnnotations(JavaCompiler.java:1170)
	at com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:856)
	at com.sun.tools.javac.main.Main.compile(Main.java:523)
```

It happens in v0.13 and v0.14-SNAPSHOT (last master).

Trying to debug it I can see that this line: 
```
final TypeElement element = context.elements.getTypeElement(customMapper.replace(".class", ""));
```
located in the `private void collectCustomMappers()` method is null. 

I really don't know what to do to solve my problem, please help !

Thank you

https://github.com/denouche/selma-bug

I cleaned the code to let the minimum source code.
But I let the hierarchy of the code, in case the problem come from this.

You just need to run `mvn clean compile` to see this:
```
[INFO] Scanning for projects...
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building test 1.0.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:3.0.0:clean (default-clean) @ test ---
[INFO] Deleting /home/denouche/workspace/selma-bug/target
[INFO] Deleting /home/denouche/workspace/selma-bug/src/main/generated-sources (includes = [], excludes = [])
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ test ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /home/denouche/workspace/selma-bug/src/main/resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.2:compile (default-compile) @ test ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 11 source files to /home/denouche/workspace/selma-bug/target/classes
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 1.429 s
[INFO] Finished at: 2016-01-15T10:41:42+01:00
[INFO] Final Memory: 16M/212M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.2:compile (default-compile) on project test: Fatal error compiling: java.lang.NullPointerException -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

Do not hesitate if you need more informations.

Thank you

--- a/processor/src/main/java/fr/xebia/extras/selma/codegen/CustomMapperWrapper.java
+++ b/processor/src/main/java/fr/xebia/extras/selma/codegen/CustomMapperWrapper.java
@@ -168,7 +168,7 @@ public class CustomMapperWrapper {
 
             for (String customMapper : customClasses) {
 
-                final TypeElement element = context.elements.getTypeElement(customMapper.replace(".class", ""));
+                final TypeElement element = context.elements.getTypeElement(customMapper.replaceAll("\\.class$", ""));
 
                 mappingMethodCount += collectCustomMethods(element, false);
 

